File: Java/1 - SimUDuck (Strategy)/src/main/java/com/mycompany/simuduck/Duck.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.simuduck;

/**
 *
 * @author ismas
 */
public abstract class Duck {
    protected FlyBehavior flyBehavior;
    protected QuackBehavior quackBehavior;
    
    public Duck() {}
    
    public abstract void display();
    
    public void performFly()
    {
        flyBehavior.fly();
    }
    
    public void performQuack()
    {
        quackBehavior.quack();
    }
    
    public void swim()
    {
        System.out.println("All ducks float, even decoys");
    }
}

File: C#/1. SimUDuck (Strategy)/Classes/Duck.cs
using SimUDuck.Interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SimUDuck.Classes
{
    public abstract class Duck
    {
        protected IFlyBehavior _FlyBehavior { get; set; }
        protected IQuackBehavior _QuackBehavior { get; set; }

        public Duck()
        {
        
        }

        public abstract void Display();

        public void PerformFly()
        {
            _FlyBehavior.Fly();
        }

        public void PerformQuack()
        {
            _QuackBehavior.Quacking();
        }


        public void Swim()
        {
            Console.WriteLine("All ducks float, evan decoys!");
        }
    }
}
